packages/providers/mgt-teamsfx-provider/src/TeamsFxProvider.ts

```diff
@@ -115,25 +60,16 @@ export class TeamsFxProvider extends IProvider {
    */
   private _accessToken: string = '';
 
-  constructor(config: TeamsFxConfig) {
+  constructor(credential: TeamsUserCredential, scopes: string[]) {
```

Comment:
It seems we're assuming the ids (clientId, initialLoginEndpoint, simpleAuthEndpoint) will always be set via TeamsFx. Is this a valid assumption here?

Reply:
Yes, that's true, all the configuration are configured by teamsfx instead of inside the provider. Because we want to make the change more smoothly when integration with graph toolkit.

For example, before using the graph toolkit, the code would be like below:

```js
   loadConfiguration({
      authentication: {
        initiateLoginEndpoint: process.env.REACT_APP_START_LOGIN_PAGE_URL,
        simpleAuthEndpoint: process.env.REACT_APP_TEAMSFX_ENDPOINT,
        clientId: process.env.REACT_APP_CLIENT_ID
      }
    });
    const credential = new TeamsUserCredential();

    const scope = [
      "User.Read",
      "User.ReadBasic.All",
    ];

   async loginBtnClick() {
    try {
      await this.credential.login(this.scope);
      this.setState({
        showLoginPage: false
      });
    } catch (err) {
      alert("Login failed: " + err);
      return;
    }
  }
```

After integration with teamsfx auth provider, user only need to add 3 lines of code without any other changes
```js
   loadConfiguration({
      authentication: {
        initiateLoginEndpoint: process.env.REACT_APP_START_LOGIN_PAGE_URL,
        simpleAuthEndpoint: process.env.REACT_APP_TEAMSFX_ENDPOINT,
        clientId: process.env.REACT_APP_CLIENT_ID
      }
    });
    const credential = new TeamsUserCredential();

    const scope = [
      "User.Read",
      "User.ReadBasic.All",
    ];

    const provider = new TeamsFxProvider(credential, scope)
    Providers.globalProvider = provider;

   async loginBtnClick() {
    try {
      await this.credential.login(this.scope);
      Providers.globalProvider.setState(ProviderState.SignedIn);
      this.setState({
        showLoginPage: false
      });
    } catch (err) {
      alert("Login failed: " + err);
      return;
    }
  }
```